pydis_site/apps/content/views/tags.py

```diff
@@ -0,0 +1,56 @@
+import re
+
+import frontmatter
+import markdown
+from django.conf import settings
+from django.http import Http404
+from django.urls import reverse
+from django.views.generic import TemplateView
+
+from pydis_site.apps.content import utils
+from pydis_site.apps.content.models import Tag
+
+COMMAND_REGEX = re.compile(r"`*!tags? (?P<name>[\w\d-]+)`*")
+
+
+class TagView(TemplateView):
+    """Handles tag pages."""
+
+    template_name = "content/tag.html"
+
+    def get_context_data(self, **kwargs) -> dict:
+        """Get the relevant context for this tag page."""
+        try:
+            tag = utils.get_tag(kwargs.get("name"))
+        except Tag.DoesNotExist:
+            raise Http404
+
+        context = super().get_context_data(**kwargs)
+        context["page_title"] = tag.name
+        body = frontmatter.parse(tag.body)
+        content = body[1]
+
+        # Check for tags which can be hyperlinked
+        start = 0
+        while match := COMMAND_REGEX.search(content, start):
```

Comment:
Would a `re.finditer` be more readable here? Rather than managing the iteration ourselves with `match.end()`.

Reply:
Yes, that's what I originally did, but how do you do the substitution? Substituting by index would not work after the first match since the index would've moved, and substituting with something like re.sub wouldn't work either since the substitution still matches, so you'll end up on the first URL again.

Reply:
I made this suggestion without even looking at what this was doing. re.sub could be used here afaik, if you make the replacement a func rather than a string.
```py
def linkify_tag_ref(match: re.Match) -> str:
    link = reverse("content:tag", kwargs={"name": match.group("name")})
    return f"[{match.group()}]({link})"

...

content = COMMAND_REGEX.search(linkify_tag_ref, content):
```

Reply:
74226511